android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -2386,22 +2386,24 @@ public void run() {
 
           if (dn <= 0) {
             if (start == end) {
-              if (!s.isEmpty() && s.charAt(0) == '\n') {
+              if (s.length() > 0 && s.charAt(0) == '\n') {
```

Comment:
ah, yeah it looks like the rest of `s` gets dropped on the floor.

Reply:
Technically, we don't support emitting text after the carriage return; we support emitting the carriage return as the last character after other text though.

* `+ 'a' > [K_ENTER] 'foo'` should never happen.

But:

* `'bar' + [K_ENTER] > 'foo' [K_ENTER]` could happen (although it will be more [verbose than that in code](https://help.keyman.com/developer/language/guide/groups#:~:text=If%20the%20final%20group%20processed%20is%20a%20context%20and%20keystroke%20group)).

Will that work at present? Do we need to open a separate issue on that?

Reply:
Will test with the example in keymanapp/help.keyman.com#346